Simplified/Reader2/Bookmarks/NYPLAnnotations.swift

```diff
@@ -216,133 +212,111 @@ import UIKit
+                             timeoutInterval: NYPLDefaultRequestTimeout)
     request.httpMethod = "GET"
     setDefaultAnnotationHeaders(forRequest: &request)
-    
+
     let dataTask = URLSession.shared.dataTask(with: request) { (data, response, error) in
-      
+
       if let error = error as NSError? {
         Log.error(#file, "Request Error Code: \(error.code). Description: \(error.localizedDescription)")
-        completionHandler(nil)
+        completion(nil)
         return
       }
       guard let data = data,
-        let json = try? JSONSerialization.jsonObject(with: data, options: []) as! [String:Any] else {
+        let jsonObject = try? JSONSerialization.jsonObject(with: data, options: []),
+        let json = jsonObject as? [String: Any] else {
           Log.error(#file, "Response from annotation server could not be serialized.")
-          completionHandler(nil)
+          completion(nil)
           return
       }
 
       guard let first = json["first"] as? [String:AnyObject],
-        let items = first["items"] as? [AnyObject] else {
+        let items = first["items"] as? [[String: AnyObject]] else {
           Log.error(#file, "Missing required key from Annotations response, or no items exist.")
-          completionHandler(nil)
+          completion(nil)
           return
       }
-      
-      for item in items {
-        guard let target = item["target"] as? [String:AnyObject],
-          let source = target["source"] as? String,
-          let motivation = item["motivation"] as? String else {
-            completionHandler(nil)
-            continue
-        }
-        
-        if source == bookID && motivation == "http://librarysimplified.org/terms/annotation/idling" {
-          
-          guard let selector = target["selector"] as? [String:AnyObject],
-            let serverCFI = selector["value"] as? String else {
-              Log.error(#file, "No CFI saved for title on the server.")
-              completionHandler(nil)
-              return
-          }
-          
-          var responseObject = [serverCFIKey : serverCFI]
-          
-          if let body = item["body"] as? [String:AnyObject],
-            let device = body["http://librarysimplified.org/terms/device"] as? String,
-            let time = body["http://librarysimplified.org/terms/time"] as? String {
-            responseObject[serverDeviceKey] = device
-            responseObject["time"] = time
-          }
-          completionHandler(responseObject)
-          return
-        }
-      }
-      Log.error(#file, "No Annotation Item found for this title.")
-      completionHandler(nil)
-      return
+
+      let readPos = items
+        .compactMap { NYPLBookmarkFactory.make(fromServerAnnotation: $0,
+                                               annotationType: .readingProgress,
+                                               bookID: bookID) }
+        .first
```

Comment:
we're creating n number of objects with compact map and are using only the first object. Using a for loop can help us avoid all this extra work.

e.g.
```
var readPos: T?
for item in items {
if let a = NYPLBookmarkFactory.make(fromServerAnnotation: item, annotationType: .readingProgress, bookID: bookID) {
readPos = a
}
```
If the items count is relatively small, then we can get away with this since it adds to code readability.

Reply:
that's kind of what I had before (a for loop) but I changed it to be more functional for readability. But yes, typically there are not hundreds of bookmarks in a book. I can change it back to a `for` if you feel strongly about it though.

Reply:
For smaller computations it's ok. Given the amount of money people pay for an iOS device, I think it should be able to handle such small computations without any hiccups.